### PR TITLE
Validate playground observations

### PR DESCRIPTION
**Pull request by AnandAili**

### CODE REVIEW

File: src/main/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommand.java [hunk 1]
```diff
@@ -13,6 +18,10 @@
     String initiativeId;
     String observer;
     Boolean smokefree;
+
+    @NotBlank(message = "Message must have at least " + SmokefreeConstants.PlaygroundObservation.MINIMUM_COMMENT_LENGTH + " character")
```
Comment (dimitriirtimid) [inline src/main/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommand.java]: this validation should be removed as the user is not obliged to enter a comment

File: src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	@Test
+	void should_notAllow_when_commentLengthExceedMaxLength() {
+		RecordPlaygroundObservationCommand recordPlaygroundObservationCommand = new RecordPlaygroundObservationCommand("123","dev-user",true, getMessage(3000));
+		Set<ConstraintViolation<RecordPlaygroundObservationCommand>> violations = validatorFactory.getValidator().validate(recordPlaygroundObservationCommand);
+		Assertions.assertEquals("Message length must not exceed "+ SmokefreeConstants.PlaygroundObservation.MAXIMUM_COMMENT_LENGTH +" characters", violations.iterator().next().getMessage());
+		Assertions.assertEquals(1, violations.size());
+
+	}
+
+	private String getMessage(int maxLength) {
```
Comment (dimitriirtimid) [inline src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java]: I think maxLength should be named 'length' here?

File: src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	void should_notAllow_when_messageContainsNotAllowedCharacters() {
+		RecordPlaygroundObservationCommand recordPlaygroundObservationCommand = new RecordPlaygroundObservationCommand("123","dev-user",true, "~");
+		Set<ConstraintViolation<RecordPlaygroundObservationCommand>> violations = validatorFactory.getValidator().validate(recordPlaygroundObservationCommand);
+		Assertions.assertEquals("Please enter only allowed special charaxters: @&(),.?\": ", violations.iterator().next().getMessage());
+		Assertions.assertEquals(1, violations.size());
+
+	}
+
+	@Test
+	void should_notAllow_when_commentIsBlank() {
```
Comment (dimitriirtimid) [inline src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java]: can be removed

File: src/main/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommand.java (hunk 1 above)
Comment (AnandAili) [reply src/main/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommand.java]: Agree. I removed it

File: src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java (hunk 2 above)
Comment (AnandAili) [reply src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java]: renamed maxLength -> length

File: src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java (hunk 3 above)
Comment (AnandAili) [reply src/test/java/io/localmotion/smokefreeplaygrounds/command/RecordPlaygroundObservationCommandTest.java]: Disabled the test case

**Pull request merged**